sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretClient.java

```diff
@@ -168,7 +203,7 @@ public Response<KeyVaultSecret> setSecretWithResponse(KeyVaultSecret secret, Con
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public KeyVaultSecret getSecret(String name) {
-        return getSecretWithResponse(name, "", Context.NONE).getValue();
+        return getSecretWithResponse(name, null, Context.NONE).getValue();
```

Comment:
Does this solve the [list secrets issue](https://github.com/Azure/azure-sdk-for-java/issues/32460) we saw a while ago?

Reply:
Maybe, we can add a test to see if `getSecret("")` results in the listing endpoint being called. But I don't think it will as that's more an issue where `getSecret` has to throw an error for `null` or `""` being the secret name as the REST API path for getting a secret is `secrets/{secretName}` and listing is `secrets/` and when the name is null or empty the path becomes `secrets`.